### ISSUE TITLE
CentOS/RHEL dependency installation docs incomplete for Fedora Server 35

### ISSUE THREAD

**Issue by Kircheneer** [nautobot/nautobot#1167]
### Environment
* Python version: 3.10.0
* Nautobot version: 1.2.0

### Steps to Reproduce
1. Install Fedora Server 35
2. Follow instructions on installing Nautobot dependencies [here](https://nautobot.readthedocs.io/en/stable/installation/centos/)
3. Follow instructions on installing Nautobot [here](https://nautobot.readthedocs.io/en/stable/installation/nautobot/)

### Expected Behavior
Nautobot installs.

### Observed Behavior
- Nautobot fails to install because the Python package `psycopg2-binary` depends on the RPM package `postgresql-devel` which is not installed as part of the dependency installation guide
```
Collecting psycopg2-binary<2.9.0,>=2.8.6
  Downloading psycopg2-binary-2.8.6.tar.gz (384 kB)
     |████████████████████████████████| 384 kB 6.8 MB/s
  Preparing metadata (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /opt/nautobot/bin/python3 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-3c3eaba8/psycopg2-binary_56be0552151b4929b40adf10ad550b02/setup.py'"'"'; __file__='"'"'/tmp/pip-install-3c3eaba8/psycopg2-binary_56be0552151b4929b40adf10ad550b02/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-jyh5o966
       cwd: /tmp/pip-install-3c3eaba8/psycopg2-binary_56be0552151b4929b40adf10ad550b02/
  Complete output (23 lines):
  running egg_info
  creating /tmp/pip-pip-egg-info-jyh5o966/psycopg2_binary.egg-info
  writing /tmp/pip-pip-egg-info-jyh5o966/psycopg2_binary.egg-info/PKG-INFO
  writing dependency_links to /tmp/pip-pip-egg-info-jyh5o966/psycopg2_binary.egg-info/dependency_links.txt
  writing top-level names to /tmp/pip-pip-egg-info-jyh5o966/psycopg2_binary.egg-info/top_level.txt
  writing manifest file '/tmp/pip-pip-egg-info-jyh5o966/psycopg2_binary.egg-info/SOURCES.txt'

  Error: pg_config executable not found.

  pg_config is required to build psycopg2 from source.  Please add the directory
  containing pg_config to the $PATH or specify the full executable path with the
  option:

      python setup.py build_ext --pg-config /path/to/pg_config build ...

  or with the pg_config option in 'setup.cfg'.

  If you prefer to avoid building psycopg2 from source, please install the PyPI
  'psycopg2-binary' package instead.

  For further information please check the 'doc/src/install.rst' file (also at
  <https://www.psycopg.org/docs/install.html>).
```
- Once the RPM package `postgresql-devel` is installed, the installation of `psycopg2-binary` fails because it depends on gcc
```
Building wheel for psycopg2-binary (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /opt/nautobot/bin/python3 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-b0ga3uoh/psycopg2-binary_c975bc9dc6404a6997c98ab988b75ad3/setup.py'"'"'; __file__='"'"'/tmp/pip-install-b0ga3uoh/psycopg2-binary_c975bc9dc6404a6997c98ab988b75ad3/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-aitad86o
       cwd: /tmp/pip-install-b0ga3uoh/psycopg2-binary_c975bc9dc6404a6997c98ab988b75ad3/
  Complete output (35 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.10
  creating build/lib.linux-x86_64-3.10/psycopg2
  copying lib/extras.py -> build/lib.linux-x86_64-3.10/psycopg2
  copying lib/_ipaddress.py -> build/lib.linux-x86_64-3.10/psycopg2
  copying lib/compat.py -> build/lib.linux-x86_64-3.10/psycopg2
  copying lib/sql.py -> build/lib.linux-x86_64-3.10/psycopg2
  copying lib/_json.py -> build/lib.linux-x86_64-3.10/psycopg2
  copying lib/errors.py -> build/lib.linux-x86_64-3.10/psycopg2
  copying lib/pool.py -> build/lib.linux-x86_64-3.10/psycopg2
  copying lib/tz.py -> build/lib.linux-x86_64-3.10/psycopg2
  copying lib/errorcodes.py -> build/lib.linux-x86_64-3.10/psycopg2
  copying lib/_lru_cache.py -> build/lib.linux-x86_64-3.10/psycopg2
  copying lib/__init__.py -> build/lib.linux-x86_64-3.10/psycopg2
  copying lib/_range.py -> build/lib.linux-x86_64-3.10/psycopg2
  copying lib/extensions.py -> build/lib.linux-x86_64-3.10/psycopg2
  running build_ext
  building 'psycopg2._psycopg' extension
  creating build/temp.linux-x86_64-3.10
  creating build/temp.linux-x86_64-3.10/psycopg
  gcc -Wno-unused-result -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -DPSYCOPG_VERSION=2.8.6 (dt dec pq3 ext lo64) -DPG_VERSION_NUM=130004 -DHAVE_LO64=1 -I/opt/nautobot/include -I/usr/include/python3.10 -I. -I/usr/include -I/usr/include/pgsql/server -c psycopg/adapter_asis.c -o build/temp.linux-x86_64-3.10/psycopg/adapter_asis.o -Wdeclaration-after-statement

  It appears you are missing some prerequisite to build the package from source.

  You may install a binary package by installing 'psycopg2-binary' from PyPI.
  If you want to install psycopg2 from source, please install the packages
  required for the build and try again.

  For further information please check the 'doc/src/install.rst' file (also at
  <https://www.psycopg.org/docs/install.html>).

  error: command 'gcc' failed: No such file or directory
  ----------------------------------------
  ERROR: Failed building wheel for psycopg2-binary
```

### Proposed solution

Have the [documentation](https://nautobot.readthedocs.io/en/stable/installation/centos/) for prerequisite installation on CentOS/RHEL include `sudo dnf install gcc postgresql-devel` which should do no harm on installations where it is included by default.

**Comment by jathanism** [comment]
Huh that's strange because the `psycopg2-binary` package is supposed to be, well, a binary package and not require compilation. It even says it in the error message:

```
  It appears you are missing some prerequisite to build the package from source.

  You may install a binary package by installing 'psycopg2-binary' from PyPI.
  If you want to install psycopg2 from source, please install the packages
  required for the build and try again.
```

I'm curious why your system isn't downloading the `.whl` vs. the tarball?

**Comment by jathanism** [comment]
@Kircheneer Can you please provide the output of `uname -a` for your system? Looking at the list of binary builds available for this package, I'm wondering if there's something that's causing yours not to match? https://pypi.org/project/psycopg2-binary/#files

**Comment by Kircheneer** [comment]
@jathanism sure!
```
# uname -a
Linux $hostname 5.14.10-300.fc35.x86_64 #1 SMP Thu Oct 7 20:48:44 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

Good point about the binary, I have no idea. It is a VM with the sole purpose of installing Nautobot onto it. Apart from the steps from the docs, nothing has been done with it.

**Comment by jathanism** [comment]
Hmmm nothing looks out of the ordinary there. There must be a quirk with the platform identification in `pip` that decides whether to do a binary install or compile. I was reading [this article on building wheels](https://realpython.com/python-wheels/#linux-building-manylinux-wheels) and if I had to guess it might be something with the `manylinux` identification.

We don't officially support Fedora, not that its' much different than RHEL. In any case, thanks for the report and your proposed solution.

**Comment by alextremblay** [comment]
psycopg uploaded binary wheels for python 3.10 on Dec 29. maybe they didn't have any py310 wheels available when you posted your issue?

If you try to reproduce your issue now, does it still happen?

**Comment by alextremblay** [comment]
Ok, I found the problem:

nautobot depends on psycopg-binary v 2.8.6, but version 2.8.6 does not have a compiled wheel for python 3.10,, as you can see here:
https://pypi.org/project/psycopg2-binary/2.8.6/#files

To fix this issue, we would need to update nautobot to use a newer version of psycopg

**Comment by glennmatthews** [comment]
Thanks for the PR! As you noted, Nautobot doesn't officially support python 3.10 yet, but we'll definitely consider adding support in version 1.3. We'll take a look at the PR soon. Thanks again!

**Comment by alextremblay** [comment]
ok, thank you

**Comment by glennmatthews** [comment]
Since this turned out to be an issue relating specifically to Python 3.10 support, I'm going to mark this as a duplicate of our catch-all issue for officially supporting 3.10 (#1268), which may be addressed in large part by PR #1255. Thank you again!
